src/components/utils/hooks/useMessagePreviews.tsx

```diff
@@ -27,29 +27,34 @@ const useMessagePreviews = () => {
   const setPreviewMessage = useMessageStore((state) => state.setPreviewMessage);
   const reset = useMessageStore((state) => state.reset);
   const { client, loading: creatingXmtpClient } = useXmtpClient();
-  const [profileIds, setProfileIds] = useState<Set<string>>(new Set<string>());
```

Comment:
Aren't we going to need both? Otherwise we are going to be adding profiles that the user isn't actually talking to to the `messageProfiles` map

Reply:
We will add them, but we shouldn't show them right? Since the key doesn't match if the profile id returned is different?

Reply:
Depends. If we iterate through that list of keys anywhere (like we do on the PreviewList) we are going to run into items that exist in the `messagePreviews` but where there are no messages.

In that case, we ignore them. But we'd have to make sure there aren't more cases to watch out for.